libraries/consul_config.rb

```diff
@@ -114,49 +116,61 @@ def tls?
             [new_resource.ca_file, new_resource.cert_file, new_resource.key_file].each do |filename|
               directory ::File.dirname(filename) do
                 recursive true
-                owner new_resource.owner
-                group new_resource.group
-                mode '0755'
+                if node['os'].eql? 'linux'
```

Comment:
Only if you're not specifying an owner and/or group (which is what we're doing). Chef produces a warning.